IPython/core/tests/test_iplib.py

```diff
@@ -118,91 +117,100 @@ def doctest_tb_verbose():
-# ---> 22         sysexit(stat, mode)
-#      23     else:
-#      24         raise ValueError('Unknown mode')
-# <BLANKLINE>
-# ...sysexit(stat, mode)
-#      10 def sysexit(stat, mode):
-# ---> 11     raise SystemExit(stat, 'Mode = %s' % mode)
-# <BLANKLINE>
-# SystemExit: (2, 'Mode = exit')
-#
-# In [23]: %xmode verbose
-# Exception reporting mode: Verbose
-#
-# In [24]: %tb
-# ---------------------------------------------------------------------------
-# SystemExit                                Traceback (most recent call last)
-# <BLANKLINE>
-# ... in <module>
-#      29     except IndexError:
-#      30         mode = 'div'
-# ---> 32     bar(mode)
-#         mode = 'exit'
-# <BLANKLINE>
-# ... in bar(mode='exit')
-#      20         except:
-#      21             stat = 1
-# ---> 22         sysexit(stat, mode)
-#         mode = 'exit'
-#         stat = 2
-#      23     else:
-#      24         raise ValueError('Unknown mode')
-# <BLANKLINE>
-# ... in sysexit(stat=2, mode='exit')
-#      10 def sysexit(stat, mode):
-# ---> 11     raise SystemExit(stat, 'Mode = %s' % mode)
-#         stat = 2
-#         mode = 'exit'
-# <BLANKLINE>
-# SystemExit: (2, 'Mode = exit')
-#     """
+def doctest_tb_sysexit():
+    """
+    In [17]: %xmode plain
+    Exception reporting mode: Plain
+
+    In [18]: %run simpleerr.py exit
+    An exception has occurred, use %tb to see the full traceback.
+    SystemExit: (1, 'Mode = exit')
+
+    In [19]: %run simpleerr.py exit 2
+    An exception has occurred, use %tb to see the full traceback.
+    SystemExit: (2, 'Mode = exit')
+
+    In [20]: %tb
+    Traceback (most recent call last):
+      File ..., in execfile
+        exec(compiler(f.read(), fname, 'exec'), glob, loc)
+      File ..., in <module>
+        bar(mode)
+      File ..., in bar
+        sysexit(stat, mode)
+      File ..., in sysexit
+        raise SystemExit(stat, f"Mode = {mode}")
+    SystemExit: (2, 'Mode = exit')
+
+    In [21]: %xmode context
+    Exception reporting mode: Context
+
+    In [22]: %tb
+    ---------------------------------------------------------------------------
+    SystemExit                                Traceback (most recent call last)
+    File ..., in execfile(fname, glob, loc, compiler)
+         70 with open(fname, 'rb') as f:
+         71     compiler = compiler or compile
+    ---> 72     exec(compiler(f.read(), fname, 'exec'), glob, loc)
+    ...<module>
+         30     except IndexError:
+         31         mode = 'div'
+    ---> 33     bar(mode)
+    <BLANKLINE>
+    ...bar(mode)
+         21         except:
+         22             stat = 1
+    ---> 23         sysexit(stat, mode)
+         24     else:
+         25         raise ValueError('Unknown mode')
+    <BLANKLINE>
+    ...sysexit(stat, mode)
+         10 def sysexit(stat, mode):
+    ---> 11     raise SystemExit(stat, f"Mode = {mode}")
+    <BLANKLINE>
+    SystemExit: (2, 'Mode = exit')
+    """
+
+
+def doctest_tb_sysexit_verbose():
```

Comment:
Is there anywhere else a test that ensures that you can call %tb multiple times with different verbosity?

Reply:
Yes, the one just before switch between plain and context. I separated this one just to make it a tiny bit easier to find the rights lines when fixing doctests.